Make sure voices get re-scheduled at the right time

tidoust opened the pull request:
Voice recordings may be slightly too long or slightly too short. Looping using the Web Audio API won't work for them since looping will take place as soon as the recording is over, meaning too early or too late.

This update makes the code use two alternating instances of buffer nodes for each voice (using two to give time to reschedule the next "play").

Note that the Web Audio API cannot re-play a buffer source that has ended, so a new node gets re-created each time.

tidoust committed:
Make sure voices get re-scheduled at the right time

Voice recordings may be slightly too long or slightly too short. Looping using
the Web Audio API won't work for them since looping will take place as soon
as the recording is over, meaning too early or too late.

This update makes the code use two alternating instances of buffer nodes for
each voice (using two to give time to reschedule the next "play").

Note that the Web Audio API cannot re-play a buffer source that has ended, so
a new node gets re-created each time.

dontcallmedom reviewed (commented):
stop no longer seems to reset the player?

tidoust committed:
Drop remaining call to obsolete ".node" property

... and no need to call "stop" anymore, which would trigger the "ended" event
at a time where we no longer want to do anything with the node.

tidoust commented:
True. Should work better now.
